src/test/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepTest.java

```diff
@@ -182,12 +183,12 @@
         story.addStep(new Statement() {
             @Override public void evaluate() throws Throwable {
                 DumbSlave s = story.j.createOnlineSlave();
-                File f1 = new File(story.j.jenkins.getRootDir(), "test.txt");
-                String fullPathToTestFile = f1.getAbsolutePath();
+                Path f1 = story.j.jenkins.getRootDir().toPath().resolve("test.txt");
+                String fullPathToTestFile = f1.toAbsolutePath().toString();
                 // Escape any \ in the source so that the script is valid
                 fullPathToTestFile = fullPathToTestFile.replace("\\", "\\\\");
-                // Ensure deleted
-                f1.delete();
+                // Ensure file does not exist
+                assertFalse(Files.exists(f1));
```

Comment:
Did the old behavior matter if this plugin's test were being run in the same workspace more than once? I guess you could use `Files.deleteIfExists(f1)` to preserve the old behavior, but I'm not sure what the original concern was.

Reply:
I too was having a hard time understanding the original behavior. Your explanation is as good as any, so we might as well preserve the existing functionality. I switched to `Files.deleteIfExists` and updated the comment to reflect your explanation.